python/libcuspatial/libcuspatial/load.py

```diff
@@ -16,8 +16,40 @@
 import ctypes
 import os
 
+# Loading with RTLD_LOCAL adds the library itself to the loader's
+# loaded library cache without loading any symbols into the global
+# namespace. This allows libraries that express a dependency on
+# this library to be loaded later and successfully satisfy this dependency
+# without polluting the global symbol table with symbols from
+# libcuspatial that could conflict with symbols from other DSOs.
+PREFERRED_LOAD_FLAG = ctypes.RTLD_LOCAL
+
+
+def _load_system_installation(soname: str):
+    """Try to dlopen() the library indicated by ``soname``
+
+    Raises ``OSError`` if library cannot be loaded.
+    """
+    return ctypes.CDLL(soname, PREFERRED_LOAD_FLAG)
+
+
+def _load_wheel_installation(soname: str):
+    """Try to dlopen() the library indicated by ``soname``
+
+    Returns ``None`` if the library cannot be loaded.
+    """
+    out = None
+    for lib_dir in ("lib", "lib64"):
```

Comment:
Can we double-check whether this loop is actually necessary? I think I originally added it because of a case where some libraries in a package were installed to lib while others were lib64 (specifically, I believe I did this for nvcomp libraries in cudf/kvikio). Our wheels should be consistent in this placement though AFAIK.

Reply:
Sure! Fastest way I could think of to test this was to just use CI, so pushed a commit: https://github.com/rapidsai/cuspatial/pull/1483/commits/b484ffde00580cb6540b6eaefcb3b3dd14612124

This path is only searched within the wheels itself, and we aren't supporting sdists, so I think we can be confident that if CI works, both paths aren't necessary.

Reply:
Just made these changes in `cudf` as well (https://github.com/rapidsai/cudf/pull/17316#issuecomment-2476823209). Now that the builds on that PR have completed, just re-triggered CI here to test one more time that `libcudf` and `libcuspatial` would play nicely together.

Reply:
Great thanks for that. Everything looks green now.